packages/transactions/src/new-signatures.ts

```diff
@@ -0,0 +1,96 @@
+import { Address, getAddressFromPublicKey } from '@solana/addresses';
+import { Decoder } from '@solana/codecs-core';
+import { getBase58Decoder } from '@solana/codecs-strings';
+import {
+    SOLANA_ERROR__TRANSACTION__FEE_PAYER_SIGNATURE_MISSING,
+    SOLANA_ERROR__TRANSACTION__SIGNATURES_MISSING,
+    SolanaError,
+} from '@solana/errors';
+import { Signature, SignatureBytes, signBytes } from '@solana/keys';
+
+import { CompilableTransaction } from './compilable-transaction';
+import { compileMessage } from './message';
+import { OrderedMap } from './ordered-map';
+import { getCompiledMessageEncoder } from './serializers';
+
+export type TransactionMessageBytes = Uint8Array & { readonly __brand: unique symbol };
+
+export interface INewTransactionWithSignatures {
+    messageBytes: TransactionMessageBytes;
+    signatures: OrderedMap<Address, SignatureBytes | null>;
+}
```

Comment:
Since I'm assuming we're gonna call the other type `TransactionMessage`, why not take that opportunity to make _this_ type the main `Transaction` type?